### ISSUE TITLE
Physical skill success formula audits

### ISSUE THREAD

**Issue by drewhershey** [sneezymud/sneezymud#163]
**Tl;Dr:**
I think there's an opportunity to improve gameplay quality and increase the benefit of character specialization if we tweak how success chances for various skills are calculated. By going through all the active skills and updating how various aspects of the attacker/victim/environment contribute to the two different types of success checks we could potentially make combat more dynamic and strategic while still maintaining a healthy level of challenge.

**Full Post**
After spending a lot of time looking through the code off and on over the last couple years, I think the way that skill success is being handled could potentially benefit from some updates. This applies specifically to physical skills, as mental skills like spells/prayers use a different type of secondary success roll (although that could stand to be reevaluated, too).

I'm curious if others feel like this is even remotely necessary or worth the time.

Currently, for physical skills there are two separate success checks. Here's my interpretation of how they work and their underlying purpose:

1) A roll within a range from 0 - 99 that more or less represents if a character knows _how_ to perform the skill.

     In other words, the chance of success based on character's understanding of the skill itself and level of experience actually executing it in practice.

2) A roll within a range from 0 - 999 (0 - 49 being a crit failure and 950 - 999 a crit success) that represents the character's chance of success against a _specific target in a specific situation_. 

     In other words, it assumes the character will execute the mechanics of the skill correctly but then determines if external factors (the surrounding environment, defensive skill of the intended victim, etc.) will still cause the attempt to fail.

     Practical example: A thief could pass the first roll and therefore perfectly execute their backstab mechanically, yet still miss if the intended victim was agile enough or the environment messed up his otherwise solid attempt. Alternatively, a victim that would normally be able to avoid the attack due to high agility might instead be hit if affected negatively enough by their environment (too dark, too loud, etc.).

Those two different checks make sense in theory. However, what appears to have happened over the years is that various factors which seem like they should be accounted for during the second, more dynamic success roll (the one that deals with the specifics of the situation) are being factored into the first roll, during what should be more of a "does the character flat out possess the knowledge/experience to perform this skill at all" check. 

Or, oftentimes, things that should be in that second roll are instead just used as criteria for hard skill failures - "you can't do that now" type responses.

The warrior bash skill is a good example. To me something like the attacker's height and weight would primarily affect their success chance against a specific victim, yet those stats are only accounted for during the first roll. 

Basically, a hobbit warrior who's mastered the art of bashing is being penalized for his low weight and short stature when determining if he knows _how_ to bash - not when he actually attempts the bash on an ogre. Something similar happens with the attacker's shield weight.

This would be a fairly big project and need to be tested carefully, but it could be done slowly over time and seems like it has potential to make the game more enjoyable. It would also provide a good opportunity to tweak the functionality of existing skills to improve their utility/variety and potentially address issues with things like the difficulty certain classes have being useful against high-level mobs.
